xarray/core/daskcompat.py

```diff
@@ -0,0 +1,519 @@
+                else (c,)
+                for s, c in zip(shape, chunks)
+            ),
+            (),
+        )
+    for c in chunks:
+        if not c:
+            raise ValueError(
+                "Empty tuples are not allowed in chunks. Express "
+                "zero length dimensions with 0(s) in chunks"
+            )
+
+    if shape is not None:
+        if len(chunks) != len(shape):
+            raise ValueError(
+                "Input array has %d dimensions but the supplied "
+                "chunks has only %d dimensions" % (len(shape), len(chunks))
+            )
+        if not all(
+            c == s or (math.isnan(c) or math.isnan(s))
+            for c, s in zip(map(sum, chunks), shape)
+        ):
+            raise ValueError(
+                "Chunks do not add up to shape. "
+                "Got chunks={}, shape={}".format(chunks, shape)
+            )
+
+    return tuple(
+        tuple(int(x) if not math.isnan(x) else np.nan for x in c) for c in chunks  # type: ignore[misc]
+    )
+
+
+def _compute_multiplier(limit: int, dtype, largest_block: int, result):
+    """
+    Utility function for auto_chunk, to fin how much larger or smaller the ideal
+    chunk size is relative to what we have now.
+    """
+    return (
+        limit
+        / dtype.itemsize
+        / largest_block
+        / math.prod(r for r in result.values() if r)
+    )
+
+
+def auto_chunks(chunks, shape, limit, dtype, previous_chunks=None):
+    """Determine automatic chunks
+    This takes in a chunks value that contains ``"auto"`` values in certain
+    dimensions and replaces those values with concrete dimension sizes that try
+    to get chunks to be of a certain size in bytes, provided by the ``limit=``
+    keyword.  If multiple dimensions are marked as ``"auto"`` then they will
+    all respond to meet the desired byte limit, trying to respect the aspect
+    ratio of their dimensions in ``previous_chunks=``, if given.
+    Parameters
+    ----------
+    chunks: Tuple
+        A tuple of either dimensions or tuples of explicit chunk dimensions
+        Some entries should be "auto"
+    shape: Tuple[int]
+    limit: int, str
+        The maximum allowable size of a chunk in bytes
+    previous_chunks: Tuple[Tuple[int]]
+    See also
+    --------
+    normalize_chunks: for full docstring and parameters
+    """
+    if previous_chunks is not None:
+        previous_chunks = tuple(
+            c if isinstance(c, tuple) else (c,) for c in previous_chunks
+        )
+    chunks = list(chunks)
+
+    autos = {i for i, c in enumerate(chunks) if c == "auto"}
+    if not autos:
+        return tuple(chunks)
+
+    if limit is None:
+        try:
+            from dask import config
+
+            # TODO plug this into configuration of other chunk managers
+            limit = config.get("array.chunk-size")
+        except ImportError:
+            limit = "128MiB"
+
+    print(f"array.chunk-size limit used = {limit}")
```

Comment:
This seems pretty janky but it's because we still need to auto-determine chunk sizes even without dask installed.

The only other option would be to make this config value (or the entire vendored function) the responsibility of the `chunkmanager`.

Reply:
i think the chunk manager should pass a `limit` kwarg if appropriate.